crates/flox/src/commands/environment.rs

```diff
@@ -600,26 +628,89 @@ pub struct Pull {
 }
 
 impl Pull {
-    pub async fn handle(self, _flox: Flox) -> Result<()> {
+    pub async fn handle(self, flox: Flox) -> Result<()> {
         subcommand_metric!("pull");
+        match self.pull_select {
+            PullSelect::New { dir, remote } => {
+                let dir = dir.unwrap_or_else(|| std::env::current_dir().unwrap());
```

Comment:
nonblocking suggestion: add some `debug!` statements so we can follow which type of pull the command resolved to.